### DISCUSSION TITLE
queryClient.invalidateQueries should clean inactive queries cache

### DISCUSSION THREAD

**Discussion by MarvelSQ** [TanStack/query#3118] [Q&A]
`queryClient.invalidateQueries` can refetch active queries. 
`{ refetchInactive: true }` will refetch immediately.
what if `queryClient.invalidateQueries` clean the cache. when queries are active, they will fetch directly and not use any cache data.
compare with `{ refetchInactive: true }`, this like a lazy update

**Comment by TkDodo** [comment]
I'm not sure I follow. react-query will never remove data from the cache unless its unused and garbage collected. It's the principle of stale-while-revalidate to show stale data until you get the new data.

What this would do is remove the data from the screen, give you a hard loading state, and then show the data again once invalidation completes.I think what you are looking for might be `queryClient.resetQueries`, which I  think does pretty much that.

**Reply by MarvelSQ** [reply]
sorry, i found there is a default setting `refetchOnMount: false` in queryClient, that's why my invalidated queries don't refetch during mount.

thank for reply.
